src/server/game/Entities/Conversation/Conversation.cpp

```diff
@@ -130,7 +131,22 @@ bool Conversation::Create(ObjectGuid::LowType lowGuid, uint32 conversationEntry,
             actorField.Type = ConversationDynamicFieldActor::ActorType::CreatureActor;
             SetDynamicStructuredValue(CONVERSATION_DYNAMIC_FIELD_ACTORS, actorsIndex++, &actorField);
         }
-        else
+        else if (!conversationTemplate->Actors.empty())
+            ++actorsIndex;
+    }
+
+    for (ObjectGuid::LowType actorGuids : conversationTemplate->ActorGuids)
+    {
+        if (actorGuids)
+        {
+            auto bounds = map->GetCreatureBySpawnIdStore().equal_range(actorGuids);
+
+            if (bounds.first == bounds.second)
```

Comment:
This isnt (yet) part of master branch but having dynamic spawning in mind, there could be multiple spawns for the same db guid and you probably should use the alive/last one